pypose/module/ukf.py

```diff
@@ -187,20 +195,24 @@ def forward(self, x, y, u, P, Q=None, R=None, t=None, k=None):
 
         xs, w = self.sigma_weight_points(x, P, k)
         xs = self.model.state_transition(xs, u, t)
-        xe = (w * xs).sum(dim=-2)
+        xe = (w * xs).sum(dim=0)
         ex = xe - xs
         P = self.compute_cov(ex, ex, w, Q)
+        if self.correct_PD_eps >= 0:
+            P = self.project_to_positive_definite(P)
 
         xs, w = self.sigma_weight_points(xe, P, k)
         ys = self.model.observation(xs, u, t)
-        ye = (w * ys).sum(dim=-2)
+        ye = (w * ys).sum(dim=0)
         ey = ye - ys
         Py = self.compute_cov(ey, ey, w, R)
 
         Pxy = self.compute_cov(ex, ey, w)
         K = Pxy @ pinv(Py)
         x = xe + bmv(K, y - ye)
         P = P - K @ Py @ K.mT
+        if self.correct_PD_eps >= 0:
+            P = self.project_to_positive_definite(P)
```

Comment:
We can define a new global function like `clamp2pd` to implement 'Positive Definite Eigenvalue Clamping' method so that other algorithms can also use it.